pkg/orchestrator/pipelines.go

```diff
@@ -80,5 +81,37 @@ func (s *PipelineOrchestrator) Delete(ctx context.Context, id string) error {
 	if len(pl.ProcessorIDs) != 0 {
 		return ErrPipelineHasProcessorsAttached
 	}
-	return s.pipelines.Delete(ctx, pl.ID)
+	return po.pipelines.Delete(ctx, pl.ID)
+}
+
+func (po *PipelineOrchestrator) UpdateDLQ(ctx context.Context, id string, dlq pipeline.DLQ) (*pipeline.Instance, error) {
+	pl, err := po.pipelines.Get(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	if pl.ProvisionedBy != pipeline.ProvisionTypeAPI {
+		return nil, cerrors.Errorf("pipeline %q cannot be updated: %w", pl.ID, ErrImmutableProvisionedByConfig)
+	}
+	// TODO lock pipeline
+	if pl.Status == pipeline.StatusRunning {
+		return nil, pipeline.ErrPipelineRunning
+	}
+
+	// cast orchestrator to a ConnectorOrchestrator to get access to the plugin
+	// config validate function
+	err = (*ConnectorOrchestrator)(po).Validate(
+		ctx,
+		connector.TypeDestination,
+		dlq.Plugin,
+		connector.Config{
+			Name:     "temporary-dlq",
```

Comment:
can we add a clarifying comment here plz? about why this name is not important, cuz I was confused at first when I saw it 👀